packages/mdc-snackbar/index.ts

```diff
@@ -21,61 +21,48 @@
  * THE SOFTWARE.
  */
 
+import {SpecificEventListener} from '@material/base';
 import {MDCComponent} from '@material/base/component';
-import MDCSnackbarFoundation from './foundation';
+import * as ponyfill from '@material/dom/ponyfill';
 import {strings} from './constants';
+import {MDCSnackbarFoundation} from './foundation';
+import {Announcer, AnnouncerFactory} from './types';
 import * as util from './util';
-import * as ponyfill from '@material/dom/ponyfill';
 
 const {
   SURFACE_SELECTOR, LABEL_SELECTOR, ACTION_SELECTOR, DISMISS_SELECTOR,
   OPENING_EVENT, OPENED_EVENT, CLOSING_EVENT, CLOSED_EVENT,
 } = strings;
 
-class MDCSnackbar extends MDCComponent {
-  static attachTo(root) {
+class MDCSnackbar extends MDCComponent<MDCSnackbarFoundation> {
+  static attachTo(root: Element) {
     return new MDCSnackbar(root);
   }
 
-  constructor(...args) {
-    super(...args);
-
-    /** @type {!HTMLElement} */
-    this.surfaceEl_;
-
-    /** @type {!HTMLElement} */
-    this.labelEl_;
+  private announce_!: Announcer; // assigned in initialize()
 
-    /** @type {!HTMLElement} */
-    this.actionEl_;
+  private actionEl_!: Element; // assigned in initialSyncWithDOM()
+  private labelEl_!: Element; // assigned in initialSyncWithDOM()
+  private surfaceEl_!: Element; // assigned in initialSyncWithDOM()
 
-    /** @type {function(!HTMLElement, !HTMLElement=): void} */
-    this.announce_;
+  private handleKeyDown_!: SpecificEventListener<'keydown'>; // assigned in initialSyncWithDOM()
+  private handleSurfaceClick_!: SpecificEventListener<'click'>; // assigned in initialSyncWithDOM()
 
-    /** @private {!Function} */
-    this.handleKeyDown_;
-
-    /** @private {!Function} */
-    this.handleSurfaceClick_;
-  }
-
-  /**
-   * @param {function(): function(!HTMLElement, !HTMLElement=):void} announceFactory
-   */
-  initialize(announceFactory = () => util.announce) {
-    this.announce_ = announceFactory();
+  initialize(announcerFactory: AnnouncerFactory = () => util.announce) {
+    this.announce_ = announcerFactory();
   }
 
   initialSyncWithDOM() {
-    this.surfaceEl_ = /** @type {!HTMLElement} */ (this.root_.querySelector(SURFACE_SELECTOR));
-    this.labelEl_ = /** @type {!HTMLElement} */ (this.root_.querySelector(LABEL_SELECTOR));
-    this.actionEl_ = /** @type {!HTMLElement} */ (this.root_.querySelector(ACTION_SELECTOR));
+    this.surfaceEl_ = this.root_.querySelector<HTMLElement>(SURFACE_SELECTOR)!;
```

Comment:
I also notice you're `!` to each querySelector call. Should we throw an error if they don't exist?